index.bs

```diff
@@ -249,69 +530,98 @@ A <dfn export>file locator</dfn> is a [=/file system locator=] whose
 whichs [=locate an entry|locates to=] a [=file entry=] |entry| that conceptually
 exists at the path `data/drafts/example.txt` relative to the root directory of
 a [=/bucket file system=],
 |locator|'s [=file system locator/kind=] has to be "{{FileSystemHandleKind/file}}",
 |locator|'s [=file system locator/path=] has to be « "`data`", "`drafts`", "`example.txt`" », and
-|locator|'s [=file system locator/root=] might include relevant identifying
-information such as the [=storage bucket=] and the disk drive.
+|locator|'s [=file system locator/file system=] has to be [=/bucket file system=].
 
 A [=/file system locator=] |a| is <dfn for="file system locator">the same locator as</dfn>
 a [=/file system locator=] |b| if
 |a|'s [=file system locator/kind=] is |b|'s [=file system locator/kind=],
-|a|'s [=file system locator/root=] is |b|'s [=file system locator/root=], and
+|a|'s [=file system locator/file system=] is |b|'s [=file system locator/file system=], and
 |a|'s [=file system locator/path=] is [=the same path as=] |b|'s [=file system locator/path=].
 
 <div algorithm>
-The <dfn export data-lt="locating an entry">locate an entry</dfn> algorithm given a
-[=/file system locator=] |locator| runs an [=implementation-defined=] series of steps adhering to
-these constraints:
 
-- If |locator| is a [=file locator=], they return a [=file entry=] or null.
-- If |locator| is a [=directory locator=], they return a [=directory entry=] or null.
-- If these steps return a non-null |entry|, then:
-  - [=Getting the locator=] with |entry| returns |locator|,
-    provided no intermediate file system operations were run.
-  - |entry|'s [=file system entry/name=] is the last [=list/item=] of |locator|'s
-    [=file system locator/path=].
+To <dfn for="file system locator" id=locator-resolve>resolve</dfn> a
+[=/file system locator=] |child| relative to a [=file system locator=] |root|:
+
+1. Let |relationship| be the result of [=file system locator/getting the relationship=]
+   between |root| and |child|.
+1. If |relationship| is equal to "`other`" or "`ancestor`", return null.
+1. If |relationship| is equal to "`self`", return « ».
+
+1. Let |childPath| be |child|'s [=file system locator/path=].
+1. Let |rootPath| be |root|'s [=file system locator/path=].
+
+1. Let |relativePath| be « ».
+1. [=list/For each=] |index| of [=the range=] from |rootPath|'s [=list/size=]
+    to |childPath|'s [=list/size=], exclusive,
+    [=list/append=] |childPath|[|index|] to |relativePath|.
+
+1. Return |relativePath|.
+
+Note: These steps have to be run on the [=file system queue=].
 
 </div>
 
 <div algorithm>
-The <dfn export data-lt="getting the locator">get the locator</dfn> algorithm given
-[=/file system entry=] |entry| runs an [=implementation-defined=] series of steps adhering to these
-constraints:
+To <dfn for="file system locator">get the relationship</dfn> between a
+[=/file system locator=] |self| to a [=file system locator=] |related|:
```

Comment:
nit: "the relationship between a *and* b" or "the relationship *from* a to b" rather than "between a to b"?